### PR TITLE
OidCollection: Use List<T> instead of LowLevelList

### PR DESCRIPTION
**Pull request by justinvp**
`System.Security.Cryptography.Encoding` already depends on `System.Collections`, so `OidCollection` can use `List<T>` instead of `LowLevelListWithIList<T>`.

Additionally, minor cleanup and added some more tests to increase coverage while making changes here.

cc: @bartonjs @AtsushiKan

### CODE REVIEW

File: src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidCollection.cs [hunk 1]
```diff
@@ -93,22 +73,10 @@ public void CopyTo(Oid[] array, int index)
             ((ICollection)this).CopyTo(array, index);
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidCollection.cs]: Can you apply a change similar to https://github.com/dotnet/corefx/pull/7247/commits/a6353946a230f0281626439306b60690655053fb ?

File: src/System.Security.Cryptography.Encoding/tests/OidCollection.cs [hunk 2]
```diff
@@ -89,5 +89,41 @@ public static void TestOidCollection()
...
+
+        [Fact]
+        public void Enumerator_Success()
+        {
+            Oid item1 = new Oid(SHA1_Oid, SHA1_Name);
+            Oid item2 = new Oid(SHA256_Oid, SHA256_Name);
+
+            OidCollection c = new OidCollection { item1, item2 };
+
+            OidEnumerator e = c.GetEnumerator();
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/tests/OidCollection.cs]: Could you also grab the IEnumerable.GetEnumerator, so it gets covered, too?

File: src/System.Security.Cryptography.Encoding/tests/OidCollection.cs [hunk 3]
```diff
@@ -89,5 +89,41 @@ public static void TestOidCollection()
 
         private const string SHA256_Name = "sha256";
         private const string SHA256_Oid = "2.16.840.1.101.3.4.2.1";
+
+        [Fact]
+        public void Enumerator_Success()
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/tests/OidCollection.cs]: It's kind of weird to have the consts in the middle.  Can you either move them to the top or bottom? (bottom would just be "move the new tests up a bit")

File: src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidEnumerator.cs [hunk 4]
```diff
@@ -16,26 +14,15 @@ internal OidEnumerator(OidCollection oids)
...
-            get
-            {
-                return _oids[_current];
-            }
-        }
+        object IEnumerator.Current => Current;
 
         public bool MoveNext()
         {
             if (_current == _oids.Count - 1)
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidEnumerator.cs]: And, how about making this `>=`, then there's more justification for editing this file :smile:

**Comment by bartonjs** [comment]
In general, looks good (even though I'm still not a huge fan of expression-bodied statements).  I think all of my comments were to get the Oid collection and AsnEncodedData collection back on an even keel.

**Comment by stephentoub** [comment]
> even though I'm still not a huge fan of expression-bodied statements

They grow on you very quickly :) Or at least they did me.

> In general, looks good

With Jeremy's feedback, LGTM2.

**Comment by JonHanna** [comment]
> They grow on you very quickly :) Or at least they did me.

My immediate response to them was that they were a pointless addition that just complicated the syntax for no real benefit and were likely to just cause confusion. Having started just with some code-it-fast throwaway code I soon found that classes with lots of single-expression members become far more readable.

File: src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidCollection.cs (hunk 1 above)
Comment (justinvp) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidCollection.cs]: Done.

File: src/System.Security.Cryptography.Encoding/tests/OidCollection.cs (hunk 2 above)
Comment (justinvp) [inline src/System.Security.Cryptography.Encoding/tests/OidCollection.cs]: Done.

File: src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidEnumerator.cs (hunk 4 above)
Comment (justinvp) [inline src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/OidEnumerator.cs]: Sure.

**Comment by justinvp** [comment]
Feedback addressed. Thanks for the reviews.

Now that `CopyTo` no longer calls the non-generic implementation, I added more tests to cover the new implementation. While doing that, I cleaned up and fleshed out the tests a little more to get to 100% line and branch coverage, and renamed the file `OidCollection.cs` -> `OidCollectionTests.cs`.

**Comment by bartonjs** [comment]
Thanks, Justin.  LGTM.

**Pull request merged**